src/components/Comment/AddComment.vue

```diff
@@ -65,16 +79,29 @@ export default {
   },
   methods: {
     save() {
-      if (!this.rating) return;
+      if (this.showRating) {
+        if (!this.rating) return;
+      }
 
-      this.onSave({rating: this.rating, comment: this.comment, user: this.signedInUser.userName})
-      .then(() => {
-        this.$emit('saved');
-        this.reset();
-        eventBus.$emit('show-toast', {body: messages.comment.commentAddSuccess, title: messages.generic.success});
-      }).catch(e => {
-        eventBus.$emit('show-toast', {body: e.message, title: messages.generic.error, type: ToastType.ERROR});
-      });
+      if (this.showRating) {
+        var payload = {
+          rating: this.rating,
+          comment: this.comment,
+          user: this.signedInUser.userName
+        }
+      } else {
+        var payload = {
+          toolId: this.toolId,
```

Comment:
there cannot be a toolId in comment component as it is a generic component. Can we pass extra params to the payload as a param to the component and add to the payload?

Reply:
i used toolId as prop but it is not required so we can still use this component wherever we want and another thing is we cannot pass it as a param